cardano-lib/default.nix

```diff
@@ -21,21 +21,78 @@ let
     };
   in builtins.toFile "topology.yaml" (builtins.toJSON topology);
 
+  mkEdgeTopologyP2P = {
+    edgeNodes ? [{addr = "127.0.0.1"; port = 3001;}]
+  , useLedgerAfterSlot ? 0
+  }:
+  let
+    mkPublicRootsAccessPoints = map (edgeNode: {address = edgeNode.addr; port = edgeNode.port;}) edgeNodes;
+    topology = {
+      localRoots = [
+        {
+           accessPoints = [];
+           advertise = false;
+           valency = 1;
+        }
+      ];
+      publicRoots = [
+        {
+          accessPoints = mkPublicRootsAccessPoints;
+          advertise = false;
+        }
+      ];
+      inherit useLedgerAfterSlot;
+    };
+  in
+    builtins.toFile "topology.yaml" (builtins.toJSON topology);
+
+  mkTopology = env: let
+    legacyTopology = mkEdgeTopology {
+      edgeNodes = [env.relaysNew];
+      valency = 2;
+      edgePort = env.edgePort or 3001;
+    };
+    p2pTopology = mkEdgeTopologyP2P {
+      inherit (env) edgeNodes;
+      useLedgerAfterSlot = env.usePeersFromLedgerAfterSlot;
+    };
+  in
+    if (env.nodeConfig.EnableP2P or false)
+    then p2pTopology
+    else legacyTopology;
+
   defaultLogConfig = import ./generic-log-config.nix;
   defaultExplorerLogConfig = import ./explorer-log-config.nix;
+  defaultDbSyncLogConfig = defaultExplorerLogConfig;
+
   mkExplorerConfig = name: nodeConfig: lib.filterAttrs (k: v: v != null) {
     NetworkName = name;
     inherit (nodeConfig) RequiresNetworkMagic;
     NodeConfigFile = "${__toFile "config-${toString name}.json" (__toJSON nodeConfig)}";
   };
 
+  mkDbSyncConfig = name: nodeConfig: (lib.filterAttrs (k: v: v != null) {
+    NetworkName = name;
+    inherit (nodeConfig) RequiresNetworkMagic;
+    NodeConfigFile = "${__toFile "config-${toString name}.json" (__toJSON nodeConfig)}";
+  })
```

Comment:
am I missing something or is
```nix
mkDbSyncConfig = name: nodeConfig: (mkExplorerConfig name nodeConfig) // defaultDbSyncLogConfig
```
?

Reply:
You're right.  The `defaultDbSyncLogConfig` alias is also unnecessary.